### ISSUE TITLE
new release version 4.2?

### ISSUE THREAD

**Issue by pabs3** [dosfstools/dosfstools#120]
There are some useful features (for eg `fatlabel --reset`) in git master and it has been quite a while since the last release. It would be great to have a new release out that could get added to the various distros that carry dosfstools.

**Comment by pali** [comment]
Hi! In git master are more fixes to processing FAT label, but there is still missing support for fixing FAT label via fsck on existing filesystem. Patch for this functionality is ready and is waiting for review in open pull request. So in current state git master branch cannot be released yet.

**Comment by wdlkmpx** [comment]
Does pali have write access to the repo?

**Comment by pali** [comment]
Yes, I have.

**Comment by kloczek** [comment]
+1 to make new release .. soon :)

**Comment by pali** [comment]
@kloczek As wrote without finishing, reviewing and merging all label patches new release is not possible. New release cannot be in unfinished/semi-working/broken state.

**Comment by Neustradamus** [comment]
+1

**Comment by zhuyan2019** [comment]
waiting for new release

**Comment by harryfan1985** [comment]
+1 waiting for new release

**Comment by pabs3** [comment]
@pali could you label the issues/pulls that need to be completed before the release can happen? Then folks could easily see what is blocking the release and potentially help out to fix those issues.

**Comment by pali** [comment]
Hello @pabs3, basically these two PRs https://github.com/dosfstools/dosfstools/pull/93 https://github.com/dosfstools/dosfstools/pull/133 and some other changes related to label checking which depends on them. Because github does not supported stacked pull requests (pull request which depends on another pull request) I planed to create it after first pull request is fixed and merged. But apparently @andreasbombe did not reacted to PR 93 since August 2018, so I do not know state.

**Comment by pabs3** [comment]
Thanks for the info, I've pinged Andreas on the OFTC IRC network,
hopefully he will respond at some point.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/

**Comment by pali** [comment]
@pabs3 do you have some updates?

**Comment by pabs3** [comment]
Sorry, I forgot to forward the response. @andreasbombe said on IRC that
they would take a look later but I guess that didn't happen yet.

-- 
bye,
pabs

https://wiki.debian.org/PaulWise

**Comment by wdlkmpx** [comment]
What about github pre-releases? 

pali should become the new project maintainer.. why can't there be 2 or 3 maintainers.

That's the circle of life.

**Comment by pali** [comment]
@wdlkmpx without #133 and #93 we should not release any new version and #93 is marked by @andreasbombe as "request changes".

And now I see that finishing #93 needs to use changes from #133. So #93 depends on #133.

**Comment by pali** [comment]
@pabs3 No, nothing happened yet.

**Comment by pali** [comment]
@pabs3 Do you have any new update?

**Comment by pabs3** [comment]
Nothing yet. At this point I suggest you proceed as if @andreasbombe is
not going to respond until they are no longer busy with other things.
So merge #133 if it is ready, try to address the comments on #93, get
other dosfstools project members to review the work and then merge it
and make a new release. If necessary you could also ask the various
dosfstools packagers to review and for Debian ask debian-devel.

https://repology.org/project/dosfstools/packages

-- 
bye,
pabs

https://wiki.debian.org/PaulWise

**Comment by kloczek** [comment]
Any progress? :P

**Comment by pali** [comment]
@kloczek: as you can see, no. no progress on #133 nor #93

**Comment by pali** [comment]
Seems that @andreasbombe does not have a time and more people are asking for a new versions. Therefore I'm going to merge #133 and #93 (even @andreasbombe marked pull requests with request changes) and then also other pending pull requests. I hope that nobody is against this action...

@pabs3: Can you take care of updating Debian dosfstools package? I would be really good to have new version in Debian 11. I'm thinking about releasing some beta/rc version with all patches which accumulated for a long time.

**Comment by pabs3** [comment]
I'll be happy to upload the new stable version once it is released if
@andreasbombe does not have the time to do so, especially if the new
version is well tested.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/

**Comment by Neustradamus** [comment]
There are a lot of PRs yet but a new release is very important, Debian 11 will freeze 2021-01-12.
Next Debian will be in 2023.
- https://github.com/dosfstools/dosfstools/pulls

Maybe more people in the team is needed to verify and merge?

**Comment by pali** [comment]
Testing and verifying code from master branch would be really useful!

**Comment by Neustradamus** [comment]
I have done a new issue for Debian 11 freeze:
- https://github.com/dosfstools/dosfstools/issues/152

**Comment by pali** [comment]
Everybody who is waiting for a new release, can you test current version from master branch if there are not any issues?

**Comment by pabs3** [comment]
Maybe make a beta release, which can be put in Debian experimental and
other places for easier testing?

-- 
bye,
pabs

https://bonedaddy.net/pabs3/

**Comment by Neustradamus** [comment]
@pali: When you create the release in GitHub, you have possibility to create a pre-release:
- https://i.ibb.co/cvkGySr/github-pre-release.png

Or you can do a draft release without a tag:
- https://i.ibb.co/wyQJqsF/github-draft-release.png

Note: I prefer a 4.2.0-beta or 4.2.0-rc tag/release:
- https://semver.org/

**Comment by pali** [comment]
I extended test suite and did more tests. Seems everything is fine, so new final version 4.2 of dosfstools is there: https://github.com/dosfstools/dosfstools/releases/tag/4.2

If you found any problem, please report a new issue on github. Closing this issue.

**Comment by Neustradamus** [comment]
@pali: Thanks and congratulation for the 4.2 release!

**Comment by pali** [comment]
@pabs3: Are you going to update a new version in Debian?

**Comment by pabs3** [comment]
I had assumed that you would do the update and I would sponsor it.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/
